projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-checkout-flow.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * Jetpack_WooCommerce_Analytics_Checkout_Flow
+ *
+ * @package automattic/jetpack
+ * @author Automattic
+ */
+
+/**
+ * Bail if accessed directly
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Jetpack_WooCommerce_Analytics_Checkout_Flow
+ * Class that handles all page view events for the checkout flow (from product view to order confirmation view)
+ */
+class Jetpack_WooCommerce_Analytics_Checkout_Flow {
+
+	use Jetpack_WooCommerce_Analytics_Trait;
+
+	/**
+	 * Jetpack_WooCommerce_Analytics_Checkout_Flow constructor.
+	 */
+	public function __construct() {
+		$this->find_cart_checkout_content_sources();
+		$this->additional_blocks_on_cart_page     = $this->get_additional_blocks_on_page( 'cart' );
+		$this->additional_blocks_on_checkout_page = $this->get_additional_blocks_on_page( 'checkout' );
+
+		// single product page view.
+		add_action( 'woocommerce_after_single_product', array( $this, 'capture_product_view' ) );
```

Comment:
I wonder if we should use hook into `wp_footer` and use `is_product()` to determine if a single product is viewed, instead of hooking into `woocommerce_after_single_product` to call `capture_product_view`. While I don't assume that `woocommerce_after_single_product` will be removed, I'm certain that `wp_footer` will never be removed.

Reply:
I kept this the same it was previously and tried not to touch how events are hooked up for existing ones (product_view, product_checkout, product_purshase).